validator/sawtooth_validator/journal/publisher.py

```diff
@@ -499,6 +506,12 @@ def queue_batch(self, batch):
         for observer in self._batch_observers:
             observer.notify_batch_pending(batch)
 
+    def can_accept_batch(self):
+        # Limit the number of batches to 2 times the publishing average.  This
+        # allows the queue to grow geometrically, if the queue is drained.
+        return len(self._pending_batches) < \
+            (2 * self._publish_count_average.value)
```

Comment:
It may be desirable for "2" to be a configurable settings (a float) so that we can do some testing for the right value to use.  Short of that, perhaps defining as a constant at the top of the file so it's easy to change for that purpose.